packages/blocks/src/__internal__/utils/url.ts

```diff
@@ -80,7 +84,58 @@ export const isValidUrl = (str: string) => {
   const url = normalizeUrl(str);
   if (url === str) {
     // Skip check if user input scheme manually
+    try {
+      new URL(url);
+    } catch {
+      return false;
+    }
     return true;
   }
   return URL_REGEX.test(url);
 };
+
+// https://en.wikipedia.org/wiki/Top-level_domain
+const COMMON_TLDS = [
```

Comment:
Perhaps we could add more domains here as listed by some statistics like [1] or [2].

[1] https://dnsinstitute.com/research/popular-tld-rank/
[2] https://w3techs.com/technologies/overview/top_level_domain

Reply:
I think the "auto identify" feature should be *conservative*, since user can always mark a piece of text as link.